storage/storage_dest.go

```diff
@@ -457,7 +487,7 @@ func (s *storageImageDestination) computeID(m manifest.Manifest) string {
 				continue
 			}
 			blobSum := m.FSLayers[i].BlobSum
-			diffID, ok := s.uncompressedOrTocDigest[blobSum]
+			diffID, ok := s.blobDiffIDs[blobSum]
```

Comment:
Actually the `copy --format …` case means that we can’t assume that TOC reuse happens _only_ with OCI schema used in `PutManifest`.

That’s, in a sense, liberating — `m.ComputeID` below can be completely TOC-unaware, and this function can then modify the computed ID itself:
```go
manifestID := m.ComputeID()
if layerWasPulledByTOCDigest {
   input := manifestID
   for _, howLayerWasPulled := range howLayersWerePulled {
     if toc := howLayerWasPulled.byThisTOCDigest; toc != N/A {
        input += "toc=" toc +","
     } else {
        input += "no-toc,"
     }
   }
   manifestID = digest(input)
}
```

---

OTOH the code looking for `DiffIDs` in `commitLayer` needs to be ready to handle non-OCI inputs, or to at least cleanly fail.

Reply:
The second part will be fixed in #2295.